eos/scattering/ee-to-ccbar.hh

```diff
@@ -0,0 +1,405 @@
+/*
+ * Copyright (c) 2021 Méril Reboud
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#ifndef EOS_GUARD_EOS_SCATTERING_EETOCCBAR_HH
+#define EOS_GUARD_EOS_SCATTERING_EETOCCBAR_HH 1
+
+#include <eos/utils/options.hh>
+#include <eos/utils/parameters.hh>
+#include <eos/utils/private_implementation_pattern.hh>
+#include <eos/utils/stringify.hh>
+#include <eos/utils/kmatrix.hh>
+#include <eos/utils/concrete-cacheable-observable.hh>
+
+#include <vector>
+
+namespace eos
+
+{
+    /*
+    Channels follow the following convention
+    #   name          type         Nf      copy
+    0   ee            PP (P)       3       -
+    1   eff(2S)       PP (P)       3       -
+    2   eff(3770)     PP (P)       3       -
+    3   eff(4040)     PP (P)       3       -
+    4   eff(4160)     PP (P)       3       -
+    5   eff(4415)     PP (P)       3       -
+    6   D0   D0bar    PP (P)       3       -
+    7   D+   D-       PP (P)       3       6 (isospin)
+    8   D0   D*0bar   VP           3       -
+    9   D*0  D0bar    VP           3       8 (c.c.)
+    10  D+   D*-      VP           3       8 (isospin)
+    11  D*+  D-       VP           3       8 (c.c.)
+    12  Ds   Ds       PP (P)       3       -
+    13  D*0  D*0bar   VV (P, S=0)  3       -
+    14  D*0  D*0bar   VV (P, S=2)  3       -
+    15  D*0  D*0bar   VV (F, S=2)  7       -
+    16  D*+  D*-      VV (P, S=0)  3       13 (isospin)
+    17  D*+  D*-      VV (P, S=2)  3       14 (isospin)
+    18  D*+  D*-      VV (F, S=2)  7       15 (isospin)
+    19  Ds+  Ds*-     VP           3       -
+    20  Ds*+ Ds-      VP           3       19 (c.c.)
+    21  Ds*+ Ds*-     VV (P, S=0)  3       -
+    22  Ds*+ Ds*-     VV (P, S=2)  3       -
+    23  Ds*+ Ds*-     VV (F, S=2)  7       -
+    */
+
+    // S -> PP channel
+    template <unsigned nchannels_, unsigned nresonances_, unsigned order_>
+    struct SPPchan :
+    public KMatrix<nchannels_, nresonances_, order_>::Channel
+    {
+
+        SPPchan(std::string name, double m1, double m2, unsigned l_orbital, std::array<Parameter, nresonances_> g0s) :
+            KMatrix<nchannels_, nresonances_, order_>::Channel(name, m1, m2, l_orbital, g0s)
+        {
+        };
+
+        //Usefull definitions for beta and rho
+        const double mm = this->_m1 - this->_m2;
```

Comment:
Why are ``mm`` and ``mp`` not regular member variables, initialized via the constructor?